placeos

```diff
@@ -0,0 +1,303 @@
+
+Start the PlaceOS environment
+
+Arguments:
+    --hard-reset            Reset the environment to a default state.
+    -s, --sentry            Set-up Sentry
+    -h, --help              Display this message
+EOF
+}
+
+start_environment() {
+    SERVICES=('')
+    hard_reset=false
+    setup_sentry=false
+    while [ ${#} -gt 0 ]; do
+        command="${1}"
+        shift
+        case ${command} in
+        --hard-reset)
+            read -p "${red}Warning:${reset} This will reset your environment. Would you like to continue (y/n)?" choice
+            case "${choice}" in
+            y | Y | YES | yes)
+                hard_reset=true
+                ;;
+            *)
+                echo "${red}exiting.${reset}"
+                exit 1
+                ;;
+            esac
+            ;;
+        -s | --sentry)
+            setup_sentry=true
+            SERVICES+=('-s')
+            ;;
+        -h | --help | help)
+            start_environment__usage
+            exit 0
+            ;;
+        *)
+            if [ -n "${command}" ]; then
+                echo "${red}Unknown option:${reset} ${command}"
+            else
+                start_environment__usage
+                exit 1
+            fi
+            ;;
+        esac
+    done
+
+    banner
+
+    # Import environment
+    set -a
+    . ./.env
+    set +a
+
+    echo "░░░ Starting PlaceOS <${PLACEOS_TAG}>"
+    echo "░░░ For detailed logging, run \`tail -f ${logfile}\`"
+
+    PLACE_DOMAIN=${PLACE_DOMAIN:-"localhost:8443"}
+    PLACE_EMAIL=${PLACE_EMAIL:-"support@place.tech"}
+    PLACE_PASSWORD=${PLACE_PASSWORD:-"development"}
+
+    BASE_DOMAIN=${PLACE_DOMAIN}
+
+    # run_or_abort \
+    #     "${base_path}/scripts/metricbeat" \
+    #     "Checking Host OS..." \
+    #     "Error occurred while checking Host OS."
+
+    run_or_abort \
+        "${base_path}/scripts/generate-secrets" \
+        "Generating secrets..." \
+        "Failed to generate secrets."
+
+    run_or_abort \
+        "${base_path}/scripts/start-services ${SERVICES[@]}" \
+        "Bringing up services..." \
+        "Failed to start services."
+
+    run_or_abort \
+        "${base_path}/scripts/init-influxdb" \
+        "Configuring InfluxDB..." \
+        "Failed to configure InfluxDB."
+
+    if [[ $setup_sentry == "true" ]]; then
+        run_or_abort \
+            "${base_path}/scripts/setup-sentry.sh" \
+            "Setting up sentry..." \
+            "Failed to setup sentry"
+
+        echo -e "\n░░░ Sentry setup complete. Login to http://$BASE_DOMAIN on port 8989 with "
+        echo "support@place.tech:development"
+    else
+        echo "░░░ To setup sentry, run \`${base_path}/scripts/setup-sentry.sh\`"
+    fi
+
+    [ ${hard_reset} == "true" ] && hard_reset
+
+    run_or_abort \
+        "${base_path}/scripts/run-init-container" \
+        "Initialising PlaceOS with default domain ($BASE_DOMAIN)..." \
+        "Failed to create user entity."
+
+    PLACEOS_DOMAIN=placeos.local:8443
```

Comment:
I don't think this loopback domain is needed any more. @w-le may be able to confirm.

Reply:
That's correct, as staff-api now supports not unverified ssl to placeos.

Also staff-api now reads the domain of incoming request URIs, and uses it in the `Host:` header for any requests to PlaceOS API, thus a static loopback domain does not need to be added to PlaceOS just for staff-api's usage anymore (and there was much rejoicing).

https://github.com/place-labs/partner-environment/blob/master/docker-compose.yml#L260-L261

Reply:
So line 174-178 can now be removed